xdem/spstats.py

```diff
@@ -9,19 +9,92 @@
 import random
 import warnings
 from functools import partial
-from typing import Callable, Union
+from typing import Callable, Union, Optional
 
 import matplotlib.pyplot as plt
 import numpy as np
 import pandas as pd
 from scipy import integrate
 from scipy.optimize import curve_fit
+from skimage.draw import disk
+from scipy.interpolate import RegularGridInterpolator, LinearNDInterpolator, griddata
+from xdem.spatial_tools import nmad
 
 with warnings.catch_warnings():
     warnings.filterwarnings("ignore", category=DeprecationWarning)
     import skgstat as skg
     from skgstat import models
 
+def robust_interp_nd_binning(df: pd.DataFrame, var_names: list[str], statistic : Union[str, Callable, None] = nmad, min_count: Optional[int] = 100) -> Callable:
```

Comment:
Would it be useful to allow `var_names` to just be a `str` in case of 1D interpolation?

Reply:
Will add!